docs/docs/mdx/importing-and-using-components.md

```diff
@@ -73,6 +73,48 @@ Because the `<Message />` and `<Chart />` components were passed into the provid
   lessonTitle="Make React components globally available as shortcodes in MDX"
 />
 
+## Lazy-loading components
+
+When you use components in your `.mdx` files, Gatsby will bundle them into the main application bundle. This can cause performance problems.
+
+In the future, [`gatsby-plugin-mdx`](/packages/gatsby-plugin-mdx) will address this. In the meantime, it can be prudent to lazy-load very large dependencies. The following snippet provides an example for lazy-loading an imaginary `Thing` component:
+
+```jsx:title=src/components/Thing/LazyThing.js
+import React from "react"
+
+import Spinner from "./somewhere"
+
+const LazyThing = props => {
+  // While the component is loading, we'll default it to using
+  // a Spinner.
+  const [Component, setComponent] = React.useState(() => Spinner)
+
+  // After the initial render, kick off a dynamic import to fetch
+  // the real component, and set it into our state.
+  React.useEffect(() => {
+    import("./Thing.js").then(Thing => setComponent(() => Thing.default))
+  }, [])
+
+  return <Component {...props} />
+}
+
+export default LazyThing
+```
+
+Inside your MDX, swap out any references to `Thing` with `LazyThing`:
+
+```diff
+-import Thing from "../components/Thing/Thing.js"
++import LazyThing from "../components/Thing/LazyThing.js"
```

Comment:
I wonder if we should keep the component as `Thing` and only change the import? This has the advantage of not changing the component usage in the MDX file.

Reply:
Yeah, I was debating this myself! In fact this is what I did in my blog when I had this problem for real. My hesitation was that it makes this thing more implicit, which might be confusing for folks on a shared codebase, or someone who forgets that it's being lazy-loaded and forgets to lazy-load their next big component.

But yeah I don't have strong opinions! @laurieontech what do you think?

Reply:
If I'm writing it myself I probably keep it as Thing. But for the benefit of drawing attention to what is happening in the docs I'd likely go for the LazyThing option.